packages/teleport/src/DesktopSession/DesktopSession.tsx

```diff
@@ -17,42 +17,59 @@ limitations under the License.
 import React from 'react';
 import styled from 'styled-components';
 import useDesktopSession, { State } from './useDesktopSession';
+import TopBar from './TopBar';
+import { Indicator, Box, Alert, Text } from 'design';
+import useTeleport from 'teleport/useTeleport';
 
 export default function Container() {
-  const state = useDesktopSession();
+  const ctx = useTeleport();
+  const state = useDesktopSession(ctx);
   return <DesktopSession {...state} />;
 }
 
 export function DesktopSession(props: State) {
-  const { setAttempt, tdpClient } = props;
+  const {
+    userHost,
+    setWsAttempt,
+    tdpClient,
+    attempt,
+    setAttempt,
+    clipboard,
+    recording,
+  } = props;
   const canvasRef = React.useRef<HTMLCanvasElement>(null);
 
   // Waits for the state hook to initialize the TdpClient.
-  // Once the client is initialized, calls for the client to connect to the
-  // server and passes in the canvas for the client to render to.
+  // Once the client is initialized, sets wsAttempt to 'success'.
   React.useEffect(() => {
-    setAttempt({ status: 'processing' });
+    setWsAttempt({ status: 'processing' });
```

Comment:
The reason its out here is that on a render event, we need to draw to the canvas, and the canvasRef necessarily lives here. My initial thought was that keeping all of the tdpClient event listeners in one place was the best design. 
```typescript
tdpClient.on('render', ({ bitmap, left, top }) => {
      const ctx = canvasRef.current.getContext('2d');
      ctx.drawImage(bitmap, left, top);
    });
```

But now that I'm thinking about it again, it would be preferable to keep all the `setXXX` state in the state hook and only house the `render` event here.